File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
n=int(input())

data=[list(map(int,input().split())) for _ in range(n)]
data.sort(key=lambda x:(x[1],x[0]))

time=data[0][1]
answer=1
for i in range(1,n):
    if time<=data[i][0]:
        time=data[i][1]
        answer+=1

print(answer)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
	int n;
	cin >> n;
	vector<pair<int, int>> Meeting_Schedule;
	int answer = 0;
	for (int i = 0; i < n; i++) {
		int start_time;
		int end_time;
		cin >> start_time >> end_time;
		Meeting_Schedule.push_back(make_pair(end_time, start_time));
	}
	sort(Meeting_Schedule.begin(), Meeting_Schedule.end());

	int time = Meeting_Schedule[0].first;
	++answer;
	for (int i = 1; i < n; i++) {
		if (time <= Meeting_Schedule[i].second) {
		    time = Meeting_Schedule[i].first;
		    answer++;
        }
	}
	cout << answer << endl;

	return 0;
}